tidb-performance-tuning-config.md

```diff
@@ -0,0 +1,527 @@
+
+    ```sql
+    SET GLOBAL tidb_max_chunk_size = 128;
+    ```
+
+- **OLAP/Analytical Workloads** (complex queries, large result sets):
+    - Recommended range: 1024-4096 rows
+    - Benefits: Improved throughput for large scans
+    - Use case: Aggregations, large table scans
+
+    ```sql
+    SET GLOBAL tidb_max_chunk_size = 4096;
+    ```
+
+### Optimize transaction mode and DML type for different workloads
+
+TiDB provides different transaction modes and DML execution types to optimize performance for various workload patterns:
+
+#### Transaction Modes
+
+- **Pessimistic Mode** (Default):
+    - Best for general workloads with potential write conflicts
+    - Provides better consistency guarantees
+
+    ```sql
+    SET SESSION tidb_txn_mode = "pessimistic";
+    ```
+
+- **Optimistic Mode**:
+    - Suitable for workloads with minimal write conflicts
+    - Better performance for multi-statement transactions
+    - Example: `begin; insert...; insert...; commit;`
+
+    ```sql
+    SET SESSION tidb_txn_mode = "optimistic";
+    ```
+
+#### DML Types
+
+`tidb_dml_type = "bulk"` indicates the bulk DML execution mode, which is suitable for scenarios where a large amount of data is written, causing excessive memory usage in TiDB.
+
+- **Pipeline DML** (New in v8.0.0):
+    - Ideal for bulk data loading with no conflicts
+    - Reduces TiDB memory usage during large writes
+    - Requirements:
+        - Auto-commit must be enabled
+        - `pessimistic-auto-commit` must be false
+
+    ```sql
+    SET SESSION tidb_dml_type = "bulk";
+    ```
+
+### Optimize group by and distinct operations with TiKV pushdown
+
+TiDB can push down aggregation operations to TiKV to optimize query performance by reducing data transfer and processing overhead. The effectiveness depends on your data characteristics:
+
+#### When to Use Pushdown
+
+- **Ideal Scenarios** (High Performance Gain):
+    - Columns with low number of distinct values (NDV)
+    - Data with high repetition (many duplicate values)
+    - Example: Status columns, category codes, date parts
+
+- **Non-Ideal Scenarios** (Potential Performance Loss):
+    - Columns with high NDV (mostly unique values)
+    - Unique identifiers or timestamps
+    - Example: User IDs, transaction IDs
+
+#### Configuration
+
+Enable pushdown optimizations at the session or global level:
+
+```sql
+-- Enable regular aggregation pushdown
+SET GLOBAL tidb_opt_agg_push_down = ON;
+
+-- Enable distinct aggregation pushdown
+SET GLOBAL tidb_opt_distinct_agg_push_down = ON;
+```
+
+### Mitigate MVCC version accumulation using in-memory engine
+
+If you observe excessive MVCC versions during performance testing (caused by either hot read/write spots or garbage collection/compaction issues), you can enable the in-memory engine feature to alleviate this problem. This feature is available since v8.4.0 and can be enabled by adding the following configuration to your TiKV configuration file.
```

Comment:
In-memory engine is introduced in v8.5.0: https://github.com/pingcap/docs/blob/f17f97ba568e350c694a14e0a8cfce1982e8d1e9/releases/release-8.5.0.md?plain=1#L152-L158